pandas/core/indexes/extension.py

```diff
@@ -259,27 +240,6 @@ def _concat_same_dtype(self, to_concat, name):
         arr = type(self._data)._concat_same_type(to_concat)
         return type(self)._simple_new(arr, name=name)
 
-    @doc(Index.take)
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
-        nv.validate_take(tuple(), kwargs)
-        indices = ensure_platform_int(indices)
-
-        taken = self._assert_take_fillable(
-            self._data,
-            indices,
-            allow_fill=allow_fill,
-            fill_value=fill_value,
-            na_value=self._na_value,
-        )
-        return type(self)(taken, name=self.name)
```

Comment:
Same here ( `self._data` vs `self._values`), and the base class just has an extra `if not self._can_hold_na:` branch

https://github.com/pandas-dev/pandas/blob/9c08fe1e192ed4ab197b042d8477f583d7a1cdeb/pandas/core/indexes/base.py#L688-L703